### PR TITLE
Fix some typing errors in DuckArrayModule

### PR DESCRIPTION
**Pull request by Illviljan**
Fixes these errors that I've been seeing locally for a while:

```python
!mypy C:\Users\J.W\Documents\GitHub\xarray\xarray\core\coordinates.py --ignore-missing-imports
C:\Users\J.W\Documents\GitHub\xarray\xarray\core\pycompat.py:48: error: Incompatible types in assignment (expression has type "Tuple[]", variable has type "Tuple[Any]")
C:\Users\J.W\Documents\GitHub\xarray\xarray\core\pycompat.py:50: error: Incompatible types in assignment (expression has type "Optional[Module]", variable has type "Optional[Literal['dask', 'pint', 'cupy', 'sparse']]")
Found 2 errors in 1 file (checked 1 source file)
```

Not sure why the CI isn't catching these?

### CODE REVIEW

File: xarray/core/pycompat.py [hunk 1]
```diff
@@ -21,12 +24,15 @@ class DuckArrayModule:
     https://github.com/pydata/xarray/pull/5561#discussion_r664815718
     """
 
-    module: ModType | None
```
Comment (headtr1ck) [inline xarray/core/pycompat.py:24]: Whoops, nice catch

File: xarray/core/pycompat.py [hunk 2]
```diff
@@ -21,12 +24,15 @@ class DuckArrayModule:
...
+    module: ModuleType | None
     version: Version
-    type: tuple[type[Any]]  # TODO: improve this? maybe Generic
```
Comment (headtr1ck) [inline xarray/core/pycompat.py:26]: I see, there was a `, ...` Missing.

**Review by headtr1ck — APPROVED** [review]
Good catch, but also no idea why CI didn't catch this. For me locally this was also not catched...

**Comment by headtr1ck** [comment]
Not sure what is up with minimum version policy and flox but it is unrelated.

**Comment by Illviljan** [comment]
I think the CI wasn't catching this because of:
https://github.com/pydata/xarray/blob/63a69fe4b2fa7a4de8a1b65826f6af4869818166/pyproject.toml#L73-L77

**Comment by headtr1ck** [comment]
> I think the CI wasn't catching this because of:
> 
> https://github.com/pydata/xarray/blob/63a69fe4b2fa7a4de8a1b65826f6af4869818166/pyproject.toml#L73-L77

Since now it seems to work, should we remove the ignore in `pyproject.toml?`

It seems that this module does mostly handle lazy duck module imports and not python version fixes.

**Comment by Illviljan** [comment]
Yeah, sounds reasonable. Maybe the reason for the ignore disappeared once DuckArrayModule was implemented?

**Pull request merged**